packages/google_maps_flutter/lib/src/location.dart

```diff
@@ -26,8 +26,9 @@ class LatLng {
   /// The longitude in degrees between -180.0 (inclusive) and 180.0 (exclusive).
   final double longitude;
 
+  /// The latitude and longitude must be adjusted because they can, in old devices, com as a double with too many digits after the dot, and, some times, the java plugin crashes because of it
   dynamic _toJson() {
-    return <double>[latitude, longitude];
+    return <double>[double.parse(latitude.toStringAsFixed(2)), double.parse(longitude.toStringAsFixed(2))];
```

Comment:
AFAIK, lat and long are 15 decimal digits right from the decimal point. We shouldn't just round to 2 digits.
If it only affects certain old android/java version, we should indicate that in the code as well. like 
```
if version < xxx {
   // do the rounding
}
```